### PR TITLE
As/include admin authentication

### PR DESCRIPTION
**Pull request by schulzalexander**
added authentication via a static key to the app, which is used for the redemption, validation and ticket data endpoints.

This is STILL IN PROGRESS until the backend side has been merged and the changes can be tested.

### CODE REVIEW

**Comment by kordianbruck** [comment]
@schulzalexander backend has been merged

File: app/src/main/java/de/tum/in/tca/ticketcheck/api/AuthenticationManager.java [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
 import de.tum.in.tca.ticketcheck.utils.Const;
+import de.tum.in.tca.ticketcheck.utils.RSASigner;
 import de.tum.in.tca.ticketcheck.utils.Utils;
 
 /**
  * This provides methods to authenticate this app installation with the tumcabe server and other instances requiring a pki.
  */
 public class AuthenticationManager {
+    private final static String ALGORITHM = "RSA";
+    private final static int RSA_KEY_SIZE = 1024;
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tca/ticketcheck/api/AuthenticationManager.java]: Any chance, we could use 4096 here?

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/model/AdminVerification.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            }
+
+            return adminVerification
+        }
+
+        fun setKeys(context: Context) {
+            Utils.setSetting(context, Const.PUBLIC_KEY, "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDnQmMnWxKL3kMd6a5MWC7sY8DBG7YkLVdfgaxr\n" +
+                    "XIE9Y25bbsSmq/1zfKLlph3j3wXm94mmxeH9rG92BcBe1MCeT4rAdYiQLjvJltk6yekZz9O4XDnR\n" +
+                    "8rvVu+9HyJVHm/ysipOeKHJreFzOxlQ2m1FWZ1djtiCEzHKEdWTQjzpc4QIDAQAB\n")
+            Utils.setSetting(context, Const.PRIVATE_KEY, "MIICdwIBADANBgkqhkiG9w0BAQEFAASCAmEwggJdAgEAAoGBAOdCYydbEoveQx3prkxYLuxjwMEb\n" +
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/model/AdminVerification.kt]: Guess this is here for testing? You should generate a new KeyPair on startup, that gets uploaded/registered on the server and then can be unlocked by an admin.

File: app/src/main/java/de/tum/in/tca/ticketcheck/utils/RSASigner.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+     * A {@link PrivateKey} instance which will be used to generate the signature.
+     */
+    private final PrivateKey privateKey;
+
+    public RSASigner(PrivateKey privateKey) {
+        this.privateKey = privateKey;
+    }
+
+    public static Signature getSignatureInstance() {
+        String signature = "SHA1WithRSA";
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tca/ticketcheck/utils/RSASigner.java]: Can we use something else? Like SHA256 maybe?

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/model/AdminVerification.kt (hunk 2 above)
Comment (schulzalexander) [reply app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/model/AdminVerification.kt]: Ok, I'll then create a new endpoint on the backend to upload the key to.

File: app/src/main/java/de/tum/in/tca/ticketcheck/utils/RSASigner.java (hunk 3 above)
Comment (schulzalexander) [reply app/src/main/java/de/tum/in/tca/ticketcheck/utils/RSASigner.java]: sure

File: app/src/main/java/de/tum/in/tca/ticketcheck/api/AuthenticationManager.java (hunk 1 above)
Comment (schulzalexander) [reply app/src/main/java/de/tum/in/tca/ticketcheck/api/AuthenticationManager.java]: sure

**Comment by schulzalexander** [comment]
@kordianbruck I built in the generation of the keys at startup, and also at startup the app will try to upload the public key to the backend if not successfully happened so far.

But we should again wait with merging until the backend part of the changes is deployed and this one can be tested.

**Comment by schulzalexander** [comment]
We tested the behaviour now for both an inactive and an active key, behaviour is as expected, i.e. access to the endpoints is only granted for activated keys.

**Review by kordianbruck — APPROVED** [review]
Looks real good.

**Pull request merged**